gptme/config.py

```diff
@@ -205,16 +164,144 @@ def get_project_config(workspace: Path | None) -> ProjectConfig | None:
         )
         # load project config
         with open(project_config_path) as f:
-            config_data = dict(tomlkit.load(f))
+            config_data = tomlkit.load(f)
 
         # Handle RAG config conversion before creating ProjectConfig
         if "rag" in config_data:
             config_data["rag"] = RagConfig(**config_data["rag"])  # type: ignore
 
-        return ProjectConfig(**config_data)  # type: ignore
+        mcp = config_data.pop("mcp", {})
+        mcp_config = MCPConfig.from_toml(mcp)
+
+        return ProjectConfig(**config_data, mcp=mcp_config)  # type: ignore
     return None
 
 
+@dataclass
+class ChatConfig:
+    # TODO: support env in chat config
+    env: dict = field(default_factory=dict)
+    # TODO: support mcp in chat config
+    mcp: MCPConfig = field(default_factory=MCPConfig)
+    model: str | None = None
+    tools: list[str] | None = None
+    tool_format: str | None = None
+    stream: bool = True
+    interactive: bool = True
+
+    @classmethod
+    def from_toml(cls, config_data: TOMLDocument) -> Self:
+        env = config_data.pop("env", {})
+        mcp = config_data.pop("mcp", {})
+        return cls(
+            env=env,
+            mcp=MCPConfig.from_toml(mcp),
+            **{k: v for k, v in config_data.items()},
+        )
+
+    def to_dict(self) -> dict:
+        """Convert ChatConfig to a clean dictionary without tomlkit objects."""
+        return {
+            "model": self.model,
+            "tools": self.tools,
+            "tool_format": self.tool_format,
+            "stream": self.stream,
+            "interactive": self.interactive,
+            "env": dict(self.env),
+            "mcp": {
+                "enabled": self.mcp.enabled,
+                "auto_start": self.mcp.auto_start,
+                "servers": [
+                    {
+                        "name": server.name,
+                        "enabled": server.enabled,
+                        "command": server.command,
+                        "args": list(server.args),
+                        "env": dict(server.env),
+                    }
+                    for server in self.mcp.servers
+                ],
+            },
+        }
+
+
+@dataclass
+class Config:
+    workspace: Path | None = None
+    user: UserConfig = field(default_factory=load_user_config)
+    project: ProjectConfig | None = None
```

Comment:
I think the workspace path should probably be stored inside the project folder.

Reply:
Actually, the workspace path is already tied to the conversation via the `workspace` dir/symlink. So it should maybe be stored within the ChatConfig. Could be stored in both, not sure if it'll be confusing. I guess the project "workspace" (more like project root or path) is not always the current working directory.

Reply:
Moved workspace into project for now as `_workspace` (not saved).